apps/emqx_bridge/src/emqx_bridge_v2.erl

```diff
@@ -1399,72 +1409,49 @@ bridge_v1_remove(
 ) ->
     Error.
 
-bridge_v1_check_deps_and_remove(BridgeV1Type, BridgeName, RemoveDeps) ->
-    BridgeV2Type = ?MODULE:bridge_v1_type_to_bridge_v2_type(BridgeV1Type),
-    bridge_v1_check_deps_and_remove(
-        BridgeV2Type,
-        BridgeName,
-        RemoveDeps,
-        lookup_conf(BridgeV2Type, BridgeName)
-    ).
-
 %% Bridge v1 delegated-removal in 3 steps:
 %% 1. Delete rule actions if RmoveDeps has 'rule_actions'
 %% 2. Delete self (the bridge v2), also delete its channel in the connector
 %% 3. Delete the connector if the connector has no more channel left and if 'connector' is in RemoveDeps
-bridge_v1_check_deps_and_remove(
-    BridgeType,
-    BridgeName,
-    RemoveDeps,
-    #{connector := ConnectorName}
-) ->
-    RemoveConnector = lists:member(connector, RemoveDeps),
-    case emqx_bridge_lib:maybe_withdraw_rule_action(BridgeType, BridgeName, RemoveDeps) of
+bridge_v1_check_deps_and_remove(BridgeV1Type, BridgeName, RemoveDeps) ->
+    BridgeV2Type = ?MODULE:bridge_v1_type_to_bridge_v2_type(BridgeV1Type),
+    case emqx_bridge_lib:maybe_withdraw_rule_action(BridgeV2Type, BridgeName, RemoveDeps) of
         ok ->
-            case remove(BridgeType, BridgeName) of
-                ok when RemoveConnector ->
-                    maybe_delete_channels(BridgeType, BridgeName, ConnectorName);
+            #{connector := ConnectorName} = lookup_conf(BridgeV2Type, BridgeName),
```

Comment:
This will change the behavior to crash if the action doesn't exist for whatever reason.  Can we ensure that's the case?  
It seems that this function is used elsewhere in the compatibility layer, maybe we could do this in a less explosive manner for now.